beacon-chain/state/state-native/getters_participation.go

```diff
@@ -39,6 +39,40 @@ func (b *BeaconState) PreviousEpochParticipation() ([]byte, error) {
 	return b.previousEpochParticipationVal(), nil
 }
 
+// CurrentEpochParticipationNoCopy corresponding to participation bits on the beacon chain without copying the data.
+// Modifications to the returned slice will modify the underlying data.
+func (b *BeaconState) CurrentEpochParticipationNoCopy() ([]byte, error) {
+	if b.version == version.Phase0 {
+		return nil, errNotSupported("CurrentEpochParticipation", b.version)
+	}
+
+	if b.currentEpochParticipation == nil {
+		return nil, nil
+	}
+
+	b.lock.RLock()
+	defer b.lock.RUnlock()
+
+	return b.currentEpochParticipation, nil
```

Comment:
```suggestion
	b.lock.RLock()
	defer b.lock.RUnlock()

	return b.currentEpochParticipation, nil
```
nitpick: The lock is acquired for such a short amount of time that I would just not bother about it

Reply:
In fact the we should always do this on every method: the above is not safe, `currentEpochParticipation` could become `nil`  from the check to the return, so the current implementation is actively not safe. So are many other methods.